privacyscore/backend/management/commands/rescanscanlist.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018 PrivacyScore Contributors
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import os
+from time import sleep
+
+from django.core.management import BaseCommand
+from django.utils import timezone
+
+from privacyscore.backend.models import Site, ScanList
+from privacyscore.utils import normalize_url
+
+
+class Command(BaseCommand):
+    help = 'Rescan all sites in exisiting ScanList.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('scan_list_id')
+        parser.add_argument('-s', '--sleep-between-scans', type=float, default=0)
+
+    def handle(self, *args, **options):
+        if not ScanList.objects.filter(id=options['scan_list_id']).exists():
+            raise ValueError('Scan List doesnot exist!')
+
+        scan_list = ScanList.objects.filter(id=options['scan_list_id'])[0]
+        sites = scan_list.sites.all()
+
+        scan_count = 0
+        for site in sites:
```

Comment:
hrm. I would expect this functionality to live on the `ScanList`. Is [`ScanList.scan`](https://github.com/PrivacyScore/PrivacyScore/blob/dcab93ee4591333d1a691f042be9fff1a320337b/privacyscore/backend/models.py#L157) not doing what you want?  The sleep parameter is probably missing there. In the future, we want to be more clever than to simply sleep for a certain time and check how many open tasks there are before adding yet another task.  Anyway, right now I think it's clever to have that functionality live in the ScanList class itself. But we can also leave it as is...

Reply:
I see this is copy and pasted from the other command. If this won't live on the ScanList, then avoid copy and pasting by refactoring the other command and then re-use existing logic. We really don't want to maintain too many places where decisions are made; esp. when it's about blacklisted sites who can make us headaches.

Reply:
I think it is there already for `ScanList` and yes sure, the sleep parameter is missing there. 
I don't know much about the COOLDOWN and BLACKLISTED logic as @malexmave added it in the `scanfromfile` command. 
I can change this to live on the `ScanList` but it will be without sleep parameter for now.

Reply:
Now that I've looked at the code more closely, I conclude that the logic isn't very important. It's merely for displaying purposes. It's not the ScanList which decides whether to scan a site or not, it's the Site itself. And it will return either `OK`, `COOLDOWN`, or `BLACKLISTED`.

So I've changed my mind about the importance of this living in central place.
I still dislike the code being copy and pasted, but I don't see much potential for refactoring. One possibility is to make [this](https://github.com/PrivacyScore/PrivacyScore/blob/dcab93ee4591333d1a691f042be9fff1a320337b/privacyscore/backend/management/commands/scanfromfile.py#L39-L45) a function which can be consumed from other modules, so that the bulk creation can eventually be done in one place only. But we can do that later... (famous last words...)

So. Feel free to leave it as is or, if you're feeling like hacking something together, extract a function from the scanfromfile command and re-use that.

Reply:
I will leave it as is for now.